dissect/target/plugins/os/unix/linux/proc.py

```diff
@@ -0,0 +1,718 @@
+        contents.readline()
+        for line in contents:
+            if not line.strip():
+                continue
+
+            socket = UnixSocket().from_line(line)
+
+            socket.stream_type_string = self.SocketStreamType(socket.type).name
+            socket.state_string = self.SocketStateType(socket.state).name
+
+            yield socket
+
+    def _parse_packet_sockets(self) -> Iterator[PacketSocket]:
+        """Internal function to parse `/proc/net/packet` entries.
+
+        Yields:
+            An iterator containing a `PacketSocket` dataclass.
+        """
+        entry = self.target.fs.path("/proc/net/packet")
+        contents = entry.open("rt")
+
+        # Skip over the header row
+        contents.readline()
+        for line in contents:
+            if not line.strip():
+                continue
+
+            socket = PacketSocket().from_line(line)
+
+            processes = self.target.proc.inode_to_pids(socket.inode)
+
+            socket.protocol_type = self.PacketProtocolTypes(socket.protocol).name
+
+            user = self.target.user_details.find(uid=socket.user)
+            user_name = user.user.name if user else str(socket.user)
+            socket.owner = user_name
+
+            for proc in processes:
+                socket.pid = proc.pid
+                socket.name = proc.name
+                socket.cmdline = proc.cmdline
+                yield socket
+
+    def _ipv6(self, addr: Union[str, int]) -> str:
+        """Convert /proc/net IPv6 hex address into standard IPv6 notation."""
+
+        addr = unpack("!LLLL", decode(addr, "hex"))
+        return IPv6Address(pack("@IIII", *addr))
+
+    def _ipv4(self, addr: Union[str, int]) -> IPv4Address:
+        """Convert /proc/net IPv4 hex address into standard IPv4 notation."""
+        if isinstance(addr, int):
+            return IPv4Address(htonl(addr))
+
+        addr = int(addr, 16)
+        return IPv4Address(htonl(addr))
+
+
+class ProcProcess:
+    def __init__(self, target: Target, pid: Union[int, str], proc_root: str = "/proc") -> None:
+        self._pid = int(pid)
+        self.target = target
+
+        # Note: ttys and pttys are not yet mapped to the process
+        if self._pid == 0:
+            # The process with PID 0 is responsible for paging and is referred to as the "swapper" or "sched" process.
+            # It is a part of the kernel and is not a regular user-mode process.
+            self.name = "swapper"
+        else:
+            try:
+                if pid == "self":
+                    self.entry = target.fs.get(fsutil.join(proc_root, "self"))
+                else:
+                    self.entry = target.fs.get(fsutil.join(proc_root, str(pid)))
+            except FileNotFoundError:
+                raise ProcessLookupError(f"Process with PID {pid} could not be found on target: {target}")
+
+            self._stat_file = self._parse_proc_stat_entry()
+            self.name = self._process_name
+
+    @internal
```

Comment:
```suggestion
    @internal
```
As this is not a `Plugin` subclass, this annotation is not needed.